tests/DataStructures/BSTreeTest.php

```diff
@@ -0,0 +1,574 @@
+            "Parent Node with key 220 should still exist. Tree was not merged correctly."
+        );
+        $this->assertNotNull(
+            $this->tree->search($leftNode),
+            "Left Node with key 225 should still exist. Tree was not merged correctly."
+        );
+        $this->assertNotNull(
+            $this->tree->search($rightNode),
+            "Parent Node with key 240 should still exist. Tree was not merged correctly."
+        );
+
+        $this->assertEquals(230, $removedNode->key, "The key of the removed node does not match");
+        $this->assertEquals('Value 230', $removedNode->value, "The value of the removed node does not match");
+        $this->assertNull(
+            $removedNode->left,
+            "The left pointer was not broken from tree correctly. Node isolation failed."
+        );
+        $this->assertNull(
+            $removedNode->right,
+            "The right pointer was not broken from tree correctly. Node isolation failed."
+        );
+        $this->assertNull(
+            $removedNode->parent,
+            "The parent pointer was not broken from tree correctly. Node isolation failed."
+        );
+    }
+
+    public function testRemoveNonExistingNode(): void
+    {
+        $this->populateTree();
+        $removedNode = $this->tree->remove(3333);
+        $this->assertNull($removedNode, "Node not found, Null should be returned.");
+    }
+
+    /**
+     * Test: Check tree size
+     */
+    public function testTreeSize(): void
+    {
+        $this->assertEquals(0, $this->tree->size());
+
+        $arrayData = [200 => "Value 200", 150 => "Value 150", 170 => "Value 170",
+            250 => "Value 250", 300 => "Value 300", 360 => "Value 360", 230 => "Value 230",
+            240 => "Value 240", 220 => "Value 220", 50 => "Value 50", 28 => "Value 28",
+            164 => "Value 164", 321 => "Value 321", 40 => "Value 40", 9 => "Value 9",
+            32 => "Value 32", 64 => "Value 64", 116 => "Value 116"
+        ];
+
+        $bsTree = new BSTree($arrayData);
+        $this->assertEquals(
+            count($arrayData),
+            $bsTree->size(),
+            "Tree size should be size of array. Failed to insert all nodes."
+        );
+    }
+
+    /**
+     * Test depth for various nodes
+     */
+    public function testGetDepth(): void
+    {
+        $this->populateTree();
+
+        $root = $this->tree->getRoot();
+
+        $node150 = $this->tree->search(150);
+        $node110 = $this->tree->search(110);
+        $node70 = $this->tree->search(70);
+
+        $this->assertEquals(0, $this->tree->getdepth($root), "The root node should have a depth of 0.");
+        $this->assertEquals(1, $this->tree->getdepth($node150), "Node 150 should have a depth of 1.");
+        $this->assertEquals(4, $this->tree->getdepth($node110), "Node 110 should have a depth of 4.");
+        $this->assertEquals(6, $this->tree->getdepth($node70), "Node 300 should have a depth of 6.");
```

Comment:
```suggestion
        $this->assertEquals(6, $this->tree->getdepth($node70), "Node 70 should have a depth of 6.");
```

Reply:
Thanks!
Just added and corrected more comments as well